docs/design/coreclr/botr/guide-for-porting.md

```diff
@@ -0,0 +1,437 @@
+    on Unix platforms. A Windows style unwind structure must be defined. In
+    addition, it is possible to enable generation of DWARF data for exposure
+    through the GDB JIT
+    <https://sourceware.org/gdb/onlinedocs/gdb/JIT-Interface.html> . This
+    support is conditional on an \#ifdef, but has been used in the past to
+    support bring up of new platforms.
+
+5.  EH Funclets. .NET requires a 2 pass exception model in order to properly
+    support exception filters. This substantially differs from the typical
+    Itanium ABI model which is used on most Linux architectures
+
+6.  OS behavior with Signals. Especially exactly where the reported instruction
+    pointer is located.
+
+7.  Little vs big endian. While.NET runtimes have been ported to big endian in
+    the past (notable examples include Mono support various game consoles, and
+    POWER, and XNA support on Xbox360) there are no current ports of CoreCLR to
+    a big endian platform.
+
+Components of the Runtime affected by a port to a new architecture
+==================================================================
+
+Notable components
+
+1.  The JIT. The jit maintains the largest concentration of architecture
+    specific logic in the stack. This is not surprising. See [Porting RyuJit](porting-ryujit.md)
+    for guidance.
+
+2.  The CLR VM. The VM is a mix of completely architecture neutral logic, and
+    very machine specific paths.
+
+3.  System.Private.CoreLib/System.Reflection. There is little to no architecture
+    specific work here that is necessary for bringup. Nice-to-have work involves
+    adding support for the architecture in the
+    System.Reflection.ImageFileMachine enum, and the ProcessorArchitecture enum,
+    and logic that manipulates it.
+
+4.  PE File format changes to add a new architecture. Also, the C# compiler likely
+    also needs a new switch to generate machine specific code for the new 
+    architecture.
+
+5.  Crossgen/Crossgen2 - As the AOT compilers that produce machine specific logic
+    from general purpose MSIL, these will be needed to improve startup performance.
+
+6.  R2RDump - This allows diagnosing issues in pre-compiled code.
+
+CLR VM
+------
+
+The VM support for architecture specific logic is encoded in a variety of
+different ways.
+
+1.  Entirely architecture specific components. These are held in an architecture
+    specific folder.
+
+2.  Features which are only enabled on certain architectures. Eg. FEATURE_HFA.
+
+3.  Adhoc \#if blocks used for specific architectures. As needed these are
+    added. The general goal is to keep these to a minimum, but difficulty here
+    is primarily driven by what special behavior the processor architecture
+    requires.
+
+My recommendation would be to look at how Arm64 is implemented in the VM for the
+most up to date model of how to implement a CPU architecture.
+
+### Architecture Specific Components
```

Comment:
There are additional important architecture specific components:
* callingconvention.h / argdestination.h
* regdisp.h
* crosscomp.h
* CONTEXT / KNONVOLATILE_CONTEXT_POINTERS / RUNTIME_FUNCTION structures in pal.h
* Unwinder for managed code (see e.g. unwinder_arm64.cpp/h) 
* gcinfodecoder.cpp/h and gcinfoencoder.cpp/h. GC info format may need to be updated too (IIRC we needed to do that for arm64 to accomodate for the larger number of registers)

Reply:
Are there also components specific to enabling debugging (and profiling or other diagnostics) that need to be called out?